core-connector/src/main/java/com/modusbox/client/router/PartiesRouter.java

```diff
@@ -80,9 +108,13 @@ public void configure() {
                         "null, " +
                         "'Output Payload: ${body}')") // default logger
                 .removeHeaders("*", "X-*")
+
+                .doCatch(CCCustomException.class,CloseWrittenOffAccountException.class)
```

Comment:
Instead of creating and throwing a custom error, you could use datasonnet together with a choice() router or a custom processor (like you have done in "GetPartyResponseValidator") to add a conditional statement